internal/cli/atlas/privateendpoints/gcp/interfaces/create.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package interfaces
+
+import (
+	"context"
+	"strings"
+
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/cli/require"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+type CreateOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	store                    store.InterfaceEndpointCreator
+	privateEndpointServiceID string
+	privateEndpointGroupID   string
+	gcpProjectID             string
+	endpoints                []string
+}
+
+func (opts *CreateOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *CreateOpts) parseEndpoints() []*atlas.GCPEndpoint {
+	endpoints := make([]*atlas.GCPEndpoint, len(opts.endpoints))
+	for i, endpoint := range opts.endpoints {
+		s := strings.Split(endpoint, "@")
```

Comment:
[nit] let's please validate this is correct, `if len(s) !=2` kind of thing, other wise if smaller you'll get a NPE when accessing `[1]`

Reply:
Agreed. I'll add a function to validate the individual endpoint strings...

Reply:
OK, I have added a new validateEndpoints function as well as a unit test for it.